sentence_transformers/cross_encoder/evaluation/CEF1Evaluator.py

```diff
@@ -0,0 +1,128 @@
+import csv
+import logging
+import os
+from typing import List
+
+import numpy as np
+from ... import InputExample
+from sklearn.metrics import f1_score
+
+logger = logging.getLogger(__name__)
+
+
+class CEF1Evaluator:
+    """
+    CrossEncoder F1 score based evaluator for binary and multiclass tasks.
+
+    The task type (binary or multiclass) is determined from the labels array.
+
+    :param sentence_pairs: A list of sentence pairs, where each pair is a list of two strings.
+    :type sentence_pairs: list[list[str]]
+    :param labels: A list of integer labels corresponding to each sentence pair.
+    :type labels: list[int]
+    :param batch_size: Batch size for prediction. Defaults to 32.
+    :type batch_size: int
+    :param show_progress_bar: Show tqdm progress bar.
+    :type show_progress_bar: bool
+    :param name: An optional name for the CSV file with stored results. Defaults to an empty string.
+    :type name: str, optional
+    :param write_csv: Flag to determine if the data should be saved to a CSV file. Defaults to True.
+    :type write_csv: bool, optional
+    """
+
+    def __init__(
+        self,
+        sentence_pairs: List[List[str]],
+        labels: List[int],
+        *,
+        batch_size: int = 32,
+        show_progress_bar: bool = None,
```

Comment:
```suggestion
        show_progress_bar: bool = False,
```
I think it's good to allow users to choose whether to show the progress bar! Most of the other evaluators use `False` here though, so for consistency it might be best to set this to False as the default.

Reply:
I've decided to set the default to None because this is the [default value](https://github.com/UKPLab/sentence-transformers/blob/20056c64c2583ff08fca8df31ff81ccc3f93b882/sentence_transformers/cross_encoder/CrossEncoder.py#L284) for CrossEncoder's predict method. This way, the decision is [delegated](https://github.com/UKPLab/sentence-transformers/blob/20056c64c2583ff08fca8df31ff81ccc3f93b882/sentence_transformers/cross_encoder/CrossEncoder.py#L317), as implemented in CrossEncoder's predict method, to the settings of a logger.

Ok, I'll switch to False, but it might be a good idea to switch to None for all evaluators in the future; not sure though.

Reply:
We can definitely reconsider this in the future indeed. For now I'm a fan of keeping the consistency :)